packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -40,15 +36,12 @@ abstract class ShutdownHooks {
   /// hooks within a given stage will be started in parallel and will be
   /// guaranteed to run to completion before shutdown hooks in the next stage are
   /// started.
-  Future<void> runShutdownHooks();
+  Future<void> runShutdownHooks(Logger logger);
```

Comment:
Could use a comment for why the Logger is passed in here and not the factory.

Reply:
done